### PR TITLE
Refactor HttpClient to not create unnecessary sockets and streams

### PR DESCRIPTION
**Pull request by ckelleyRH**
The current implementation creates a set of null streams and sockets, then decides part way through the send process whether to use an SSL socket or not, creating streams from the appropriate socket accordingly. This is incompatible with using try-with-resources as you cannot reassign to a controlled resource. Also, it is wasteful and confusing that the unnecessary objects exist (but understandable as the code predates the existence of try-with-resources).

Separate SSL/non-SSL send methods are created, which only create the necessary streams and sockets for that operation. New helper methods are introduced to send/handle the request/response.

### CODE REVIEW

**Comment by ckelleyRH** [comment]
The Azure failure is a symlinking issue due to a difference in how Maven and `rpmbuild` link the JARs, it is unrelated to this change.

File: base/tools/src/main/java/com/netscape/cmstools/HttpClient.java [hunk 1]
```diff
@@ -151,96 +165,85 @@ public void send(String ifilename, String ofilename, String tokenName, String db
...
 
-            bs = new ByteArrayOutputStream();
-            ps = new PrintStream(bs);
             ps.print(Utils.base64encode(bout, true));
+
             System.out.println(bs.toString());
 
             System.out.println("");
             System.out.println("The response in binary format is stored in " + ofilename);
             System.out.println("");
```
Comment (edewata) [inline base/tools/src/main/java/com/netscape/cmstools/HttpClient.java:245]: In the original code any exception that happens in this part of the code will be caught and rethrown. In the new code the exception will get swallowed. Should we just rethrow the exception?

Comment (ckelleyRH) [reply base/tools/src/main/java/com/netscape/cmstools/HttpClient.java:245]: Good spot, I dropped the empty `catch` and let the calling code deal with the exception.

**Review by edewata — APPROVED** [review]
LGTM, thanks for the update! The JSS issue has been fixed as well.

**Comment by ckelleyRH** [comment]
Thanks @edewata @fmarco76 !

**Pull request merged**